### ISSUE TITLE
set page inside controller

### ISSUE THREAD

**Issue by vagyriag** [impresspages/ImpressPages#727]
Hi!

Couldn't find a way to set the page from inside a controller, after the action and parametters get resolved, I'll be extremely useful for widget management in different plugin routes.

Something a little more flexible than this page setting
$routes['plugin/{id}'] = array(
    ...
    'page' => ipContent()->getPage('39') // all get the same page
);

Thanks.

**Comment by maskas** [comment]
There is nothing better than 'page' setting. Because to use widget, you have to have a real page in the database. If you like, you can create page on the fly instead of using menu.

**Comment by vagyriag** [comment]
yeah, I understand that and I'm creating a page in every Grid afterCreate inside my AdminController, so I have a specific related page for every entry, but I can't set the page to that one once the PublicController route is visited

**Comment by maskas** [comment]
I feel like I'm still not on the same page. But you can create pages on the fly when you create a record in GRID.

**Comment by vagyriag** [comment]
yeah, that's what I'm doing, I create a new page for every record just like you just said.
But then I need to set the page that is related to that record, something like

```
$routes['plugin/{id}'] = array(
    ...
    'page' => ipContent()->getPage('myCreatedPage'.$id) // a page that is already created
);
```

so when I go to *'/plugin/1'* it'll set the page **'myCreatedPage1'**, and with *'/plugin/2'* it'll be **'myCreatedPage2'** and so on...


I supose that could be a something like 
```
class PublicController extends \Ip\Controller {
    public function plugin ( $id ) {
         ipContent()->setPage('myCreatedPage'.$id);
         ...
    }
}
```

is there a way to acomplish that?

**Comment by maskas** [comment]
You probably can catch ipRouteAction job to do so. But it will be more complex. And I don't see much benefit of it. There were idea to put this thing inside a controller. Haven't been implemented yet.

**Comment by vagyriag** [comment]
thanks
you think that could be some downside to that implementation?

**Comment by maskas** [comment]
You mean the downside of having setPage in the controller? I'm not sure. There could be some chicken / egg issues. Now pageId is known much earlier before the controller is executed. But I agree 100% that it would be handsome.
